src/napari_epitools/_widget.py

```diff
@@ -173,10 +173,33 @@ def run() -> npt.NDArray[np.float64]:
     return run()
 
 
+def _init_segmentation_widget(widget):
+    """Add callbacks for the widget"""
+
+    viewer = napari.current_viewer()
+
+    # Automatically select a newly added Image layer
+    viewer.layers.events.inserted.connect(
+        lambda event: _select_inserted_image(event.value, widget.input_image),
+    )
+
+
+def _select_inserted_image(new_layer, widget):
+    """Update the selected Image when a image layer is added"""
+
+    if not isinstance(new_layer, napari.layers.Image):
+        return
+
+    widget.native.setCurrentIndex(
+        len(widget) - 1
+    )  # the new layer is always last in the list
```

Comment:
Should go above? One benefit is this won't be over 3 lines

Reply:
yep definitely, much more readable